Formula/dotenv-linter.rb

```diff
@@ -0,0 +1,39 @@
+class DotenvLinter < Formula
+  desc "Lightning-fast linter for .env files written in Rust"
+  homepage "https://dotenv-linter.github.io"
+  url "https://github.com/dotenv-linter/dotenv-linter/archive/v2.2.1.tar.gz"
+  sha256 "0ccf8f221a84c935bb885b863ba54283cc26a9724aae6a15766a387ccc4d3f4d"
+  license "MIT"
+  head "https://github.com/dotenv-linter/dotenv-linter.git"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    checks = shell_output("#{bin}/dotenv-linter --show-checks").split("\n")
+    assert_includes checks, "DuplicatedKey"
+    assert_includes checks, "UnorderedKey"
+    assert_includes checks, "LeadingCharacter"
+
+    (testpath/".env").write <<~EOS
+      FOO=BAR
+      FOO=BAR
+      BAR=FOO
+    EOS
+    (testpath/".env.test").write <<~EOS
+      1FOO=BAR
+      _FOO=BAR
+    EOS
+    expected = <<~EOS
+      .env:2 DuplicatedKey: The FOO key is duplicated
+      .env:3 UnorderedKey: The BAR key should go before the FOO key
+      .env.test:1 LeadingCharacter: Invalid leading character detected
+
+      Found 3 problems
+    EOS
+    assert_equal expected, shell_output("#{bin}/dotenv-linter", 1)
```

Comment:
I think it would be nicer to be a little more lenient here and do things like `assert_match "DuplicateKey", output` to make sure we don't need to update the test for whitespace changes upstream.